jdaviz/configs/default/plugins/model_fitting/initializers.py

```diff
@@ -7,21 +7,40 @@
 """
 import numpy as np
 
+import astropy.modeling.models as models
+from astropy import units as u
+from jdaviz.models import BlackBody
+
 __all__ = [
-    'initialize'
+    'initialize',
+    'get_model_parameters'
 ]
 
 AMPLITUDE = 'amplitude'
 POSITION = 'position'
 WIDTH = 'width'
 
-model_parameters = {"Gaussian1D": ["amplitude", "stddev", "mean"],
-                    "Const1D": ["amplitude"],
-                    "Linear1D": ["slope", "intercept"],
-                    "PowerLaw1D": ["amplitude", "x_0", "alpha"],
-                    "Lorentz1D": ["amplitude", "x_0", "fwhm"],
-                    "Voigt1D": ["x_0", "amplitude_L", "fwhm_L", "fwhm_G"],
-                    }
+MODELS = {
+     'Const1D': models.Const1D,
+     'Linear1D': models.Linear1D,
+     'Polynomial1D': models.Polynomial1D,
+     'Gaussian1D': models.Gaussian1D,
+     'Voigt1D': models.Voigt1D,
+     'Lorentz1D': models.Lorentz1D,
+     'BlackBody': BlackBody
+     }
+
+
+def get_model_parameters(model_cls, model_kwargs={}):
```

Comment:
I think this function should be tested on its own for all the models we support. Just in case something changes over at `astropy` that would break this.

Reply:
What do you think we should test?  Just that the function runs or that the model parameter names match the old dictionary?  My hope was that this would be more general and would generalize the code, make it easier to implement new models, and not break if parameter names change in astropy.  But I suppose the mappings in `initializers._p_names` would now become the breaking point for any name changes in astropy.

Reply:
Though rare, there was precedent of renaming a parameter or adding/removing existing one before.

If there is already tests that cover fitting all the supported models here, I guess that is good enough. Otherwise, will checking if this function returns the hardcoded stuff that you remove defeat the purpose of removing them?

Reply:
added a test.  I don't think it defeats the purpose as we'll now be made aware when there is any change.  Although I only loop through the "expected" dictionary from before, rather than vice versa (so the test becomes one more place to edit when adding a new model in order for that model to be covered by testing).  If you think this should be flipped to catch when a model is added to the code but not the tests, I can change it to that easily enough.